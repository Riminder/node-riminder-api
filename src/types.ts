export interface ProfilesOptions {
  source_ids: Array<string>;
  date_start: Date;
  date_end: Date;
  page: number;
  seniority?: string;
  filter_id?: string;
  filter_reference?: string;
  stage?: string;
  rating?: string;
  limit?: number;
  sort_by?: string;
  order_by?: string;
}

interface ProfileOptionId {
  source_id: string;
  profile_id: string;
}

interface ProfileOptionReference {
  source_id: string;
  profile_reference: string;
}

export type ProfileOptionIdOrReference = ProfileOptionId | ProfileOptionReference;

interface TrainingMetadataId {
  filter_id: string;
  stage: string;
  stage_timestamp: Date;
  rating: number;
  rating_timestamp: Date;
}

interface TrainingMetadataReference {
  filter_reference: string;
  stage: string;
  stage_timestamp: Date;
  rating: number;
  rating_timestamp: Date;
}

export interface ProfileUpload {
  source_id: string;
  profile_reference: string;
  timestamp_reception: Date;
  training_metadata: Array<TrainingMetadataId | TrainingMetadataReference>;
}

interface StagePatchProfileIdFilterId {
  source_id: string;
  profile_id: string;
  filter_id: string;
  stage: string | null;
}

interface StagePatchProfileIdFilterReference {
  source_id: string;
  profile_id: string;
  filter_reference: string;
  stage: string | null;
}

interface StagePatchProfileReferenceFilterId {
  source_id: string;
  profile_reference: string;
  filter_id: string;
  stage: string | null;
}

interface StagePatchProfileReferenceFilterReference {
  source_id: string;
  profile_reference: string;
  filter_reference: string;
  stage: string | null;
}

export type StagePatch =
  StagePatchProfileIdFilterId |
  StagePatchProfileIdFilterReference |
  StagePatchProfileReferenceFilterId |
  StagePatchProfileReferenceFilterReference;