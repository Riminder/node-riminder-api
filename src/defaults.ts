let defaults: any = {
  API_URL: "https://www.riminder.net/sf/public/api/v1.0",
  API_SECRET: null,
  API_Key: null
};

export default defaults;