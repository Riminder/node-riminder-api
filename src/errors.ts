import { RiminderAPIResponse } from "./types";

export class APIError extends Error {
    response: RiminderAPIResponse;
    constructor(message: string, response: RiminderAPIResponse)Â {
        super(message);
        this.response = response;
    }
}