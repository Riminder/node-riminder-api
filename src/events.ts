export default [
  "profile.parse.success",
  "profile.parse.error",
  "profile.score.success",
  "profile.score.error",
  "filter.train.start",
  "filter.train.success",
  "filter.train.error",
  "filter.score.start",
  "filter.score.success",
  "filter.score.error"
];