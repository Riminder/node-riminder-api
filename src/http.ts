import "whatwg-fetch";
import "./defaults";
import defaults from "./defaults";
import { Riminder } from "./index";

declare interface RiminderAPIResponse {
  code: number;
  message: string;
  data?: any;
}

export const httpRequest = (url: string, options?: any) => {
  let headers = {
    "X-API-Key": Riminder._instance.API_Key || defaults.API_Key,
  };

  let opts = {
    headers,
    credentials: "include",
    ...options
  };

  return fetch(url, opts)
    .then(successHandler, errorHandler)
    .then((json: RiminderAPIResponse) => json.data);
};

const successHandler = (response: Response) => {
  if (response.status === 200 || response.status === 201) {
    return response.json();
  }
  console.error(response);
};

const errorHandler = (err: any) => {
  let error = new Error(err.message);
  (<any>error).response = err;
  throw error;
};