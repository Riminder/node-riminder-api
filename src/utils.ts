import "./defaults";
import { httpRequest } from "./http";
import defaults from "./defaults";

export const create = (object: string, data: any) => {
  let url = `${defaults.API_URL}/${object}`;
  let body = JSON.stringify(data);
  let options = {
    method: "POST",
    body,
    headers: {
      "Content-Type": "application/json"
    }
  };
  return httpRequest(url, options);
};

export const getOne = (object: string, id: string) => {
  let url = `${defaults.API_URL}/${object}/${id}`;
  return httpRequest(url);
};

export const getList = (object: string, options: any) => {
  let whereClause = options.where || {};
  let page = options.page;
  let itemsPerPage = options.itemsPerPage;
  let urlParams = generateURLParams(whereClause, page, itemsPerPage);
  let url = `${defaults.API_URL}/${object}${urlParams ? `?${urlParams}` : ""}`;
  return httpRequest(url);
};

export const generateURLParams = (whereClause: any, page?: number, itemsPerPage?: number, sort?: string) => {
  let keys = Object.keys(whereClause);
  let URLParams = "";
  for (let key of keys) {
    URLParams = `${URLParams}&${key}=${whereClause[key]}`;
  }
  URLParams = `${URLParams}${page ? `&page=${page}` : ""}${itemsPerPage ? `&limit=${itemsPerPage}` : ""}${sort ? `&sort_by=${sort}` : ""}`;
  return URLParams.slice(1);
};