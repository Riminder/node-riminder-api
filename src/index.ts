import { httpRequest } from "./http";
import { create, getOne, getList } from "./utils";
import objects from "./objects";

export interface RiminderOptions {
  API_Key: string;
  API_Secret?: string;
}

export class Riminder {
  static _instance: Riminder;
  public API_Key: string;
  public objects: any;
  public webhooks: any;
  constructor(options: RiminderOptions) {
    if (Riminder._instance) {
      let error = new Error("You can not instanciate more than one instance of Riminder SDK");
      throw error;
    }

    if (!options.API_Key) {
      let error = new Error("No API Key was supplied for Riminder SDK");
      throw error;
    }

    this.API_Key = options.API_Key;
    this._init();
    Riminder._instance = this;
  }

  private _init() {
    this.objects = objects;
    this.webhooks = {};
  }

}