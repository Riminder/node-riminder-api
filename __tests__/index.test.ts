import * as fs from "fs";
import { generateURLParams } from "../src/utils";
import { httpRequest } from "../src/http";
import Riminder = require("../src/index");
import { RiminderOptions, ProfilesOptions, ProfileUpload, StagePatch, Stage, RatingPatch, Seniority, SortBy, OrderBy } from "../src/types";

let app: Riminder;

function getQueryParamAsArray(query: string): Array<string> {
  return query
    .split("&")
    .map((entry: string) => entry.split("=")[0]);
}

describe("Other tests", () => {
  describe("\"Riminder\" object relative tests", () => {
    test("it should throw an error when no key is given", () => {
      expect(() => {
        app = new Riminder({} as RiminderOptions);
      }).toThrowErrorMatchingSnapshot();
    });
  });

  describe("utils module relative tests", () => {
    test("This should return null if there is no input data", () => {
      expect(generateURLParams(null)).toBeNull();
    });
  });

  describe("http module relative tests", () => {
    test("It should throw an error if the error code is not 200 or 201", () => {
      const headers = {
        "X-API-Key": "api_key",
      };

      expect.assertions(1);
      return httpRequest("localhost", { error: true, headers }).catch((e) => {
        expect(e).toMatchSnapshot();
      });
    });

    test("it should throw an error if something went wrong with the request", () => {
      const headers = {
        "X-API-Key": "api_key",
      };
      expect.assertions(1);
      return httpRequest("localhost", { reject: true, headers }).catch((e) => {
        expect(e).toMatchSnapshot();
      });
    });
  });
});

describe("Wrapper test", () => {
    app = new Riminder({API_Key: "api_key"});
    describe("Source endpoints", () => {
        test("It should call the get source list endpoint", () =>
        app.source.list().then((response: any) => {
          expect(response).toMatchSnapshot();
        }));

        test("It should call the get source endpoint", () => {
          app.source.get("id").then((response: any) => {
            expect(response).toMatchSnapshot();
          });
        });
      });

    describe("Filter endpoints", () => {
      test("It should call the get filter list endpoint", () => {
        app.filter.list()
          .then((response: any) => {
            expect(response).toMatchSnapshot();
          });
      });

      test("It sould call the get filter endpoint with the filter id", () => {
        const options = {
          filter_id: "filter_id"
        };
        app.filter.get(options)
          .then((response: any) => {
            expect(response).toMatchSnapshot();
          });
      });

      test("It sould call the get filter endpoint with the filter reference", () => {
        const options = {
          filter_reference: "filter_reference"
        };
        app.filter.get(options)
          .then((response: any) => {
            expect(response).toMatchSnapshot();
          });
      });
    });

    describe("Profile endpoints", () => {
      test("It should call the get profile endpoint using Date object", () => {
        const options: ProfilesOptions = {
          source_ids: ["source1", "source2"],
          date_start: new Date(0),
          date_end: new Date(1234),
          page: 1,
          seniority: Seniority.ALL,
          filter_id: "filter_id",
          filter_reference: "filter_reference",
          stage: Stage.YES,
          rating: 1,
          limit: 30,
          sort_by: SortBy.RECEPTION,
          order_by: OrderBy.DESC,
        };
        app.profile.list(options)
          .then((response: any) => {
          expect(response).toMatchSnapshot();
          expect(getQueryParamAsArray(response.url.query)).toEqual(Object.keys(options));
        });
      });

      test("It should call the get profile endpoint using Date number", () => {
        const options: ProfilesOptions = {
          source_ids: ["source1", "source2"],
          date_start: 0,
          date_end: (new Date("2018-01-01")).getTime(),
          page: 1,
          seniority: Seniority.ALL,
          filter_id: "filter_id",
          filter_reference: "filter_reference",
          stage: Stage.YES,
          rating: 1,
          limit: 30,
          sort_by: SortBy.RECEPTION,
          order_by: OrderBy.DESC,
        };
        app.profile.list(options)
          .then((response: any) => {
          expect(response).toMatchSnapshot();
          expect(getQueryParamAsArray(response.url.query)).toEqual(Object.keys(options));
        });
      });

      test("It should call the get profile endpoint with the profile id", () => {
        const options = {
          source_id: "source_id",
          profile_id: "profile_id",
        };
        app.profile.get(options).then((response: any) => {
          expect(response).toMatchSnapshot();
        });
      });

      test("It should call the get profile endpoint with the profile reference", () => {
        const options = {
          source_id: "source_id",
          profile_reference: "profile_reference",
        };
        app.profile.get(options).then((response: any) => {
          expect(response).toMatchSnapshot();
        });
      });

      test("It should call the get profile document endpoint with the profile id", () => {
        const options = {
          source_id: "source_id",
          profile_id: "profile_id",
        };
        app.profile.document.list(options).then((response: any) => {
          expect(response).toMatchSnapshot();
        });
      });

      test("It should call the get profile document endpoint with the profile reference", () => {
        const options = {
          source_id: "source_id",
          profile_reference: "profile_reference",
        };
        app.profile.document.list(options).then((response: any) => {
          expect(response).toMatchSnapshot();
        });
      });

      test("It should call the get profile parsing endpoint with the profile id", () => {
        const options = {
          source_id: "source_id",
          profile_id: "profile_id",
        };
        app.profile.parsing.get(options).then((response: any) => {
          expect(response).toMatchSnapshot();
        });
      });

      test("It should call the get profile parsing endpoint with the profile reference", () => {
        const options = {
          source_id: "source_id",
          profile_reference: "profile_reference",
        };
        app.profile.parsing.get(options).then((response: any) => {
          expect(response).toMatchSnapshot();
        });
      });

      test("It should call the get profile scoring endpoint with the profile id", () => {
        const options = {
          source_id: "source_id",
          profile_id: "profile_id",
        };
        app.profile.scoring.list(options).then((response: any) => {
          expect(response).toMatchSnapshot();
        });
      });

      test("It should call the get profile scoring endpoint with the profile reference", () => {
        const options = {
          source_id: "source_id",
          profile_reference: "profile_reference",
        };
        app.profile.scoring.list(options).then((response: any) => {
          expect(response).toMatchSnapshot();
        });
      });

      test("It should call the post resume endpoint", () => {
        const data: ProfileUpload = {
          source_id: "source_id",
          profile_reference: "ref",
          timestamp_reception: new Date(Date.now()),
          training_metadata: [{
            filter_id: "filter_id1",
            stage: Stage.YES,
            stage_timestamp: new Date(Date.now()),
            rating: 2,
            rating_timestamp: new Date(Date.now())
          },
          {
            filter_reference: "filter_reference",
            stage: Stage.YES,
            stage_timestamp: new Date(Date.now()),
            rating: 2,
            rating_timestamp: new Date(Date.now())
          }],
        };

        const file = fs.createReadStream("./test.txt");
        app.profile.add(data, file)
          .then((response: any) => {
            const responseWithoutBody = {
              url: response.url,
              options: {
                headers: response.options.headers,
                method: response.options.method
              }
            };
            expect(responseWithoutBody).toMatchSnapshot();
        });
      });

      test("It should call the patch stage endpoint with the the profile id and filter id", () => {
        const data: StagePatch = {
          source_id: "source_id",
          profile_id: "profile_id",
          filter_id: "filter_id",
          stage: Stage.YES,
        };

        app.profile.stage.set(data)
          .then((response: any) => {
            const responseWithoutBody = {
              url: response.url,
              options: {
                headers: response.options.headers,
                method: response.options.method
              }
            };
            expect(responseWithoutBody).toMatchSnapshot();
          });
      });

      test("It should call the patch stage endpoint with the the profile id and filter reference", () => {
        const data: StagePatch = {
          source_id: "source_id",
          profile_id: "profile_id",
          filter_id: "filter_id",
          stage: Stage.YES,
        };
        app.profile.stage.set(data)
          .then((response: any) => {
            const responseWithoutBody = {
              url: response.url,
              options: {
                headers: response.options.headers,
                method: response.options.method
              }
            };
            expect(responseWithoutBody).toMatchSnapshot();
          });
      });

      test("It should call the patch stage endpoint with the the profile reference and filter id", () => {
        const data: StagePatch = {
          source_id: "source_id",
          profile_id: "profile_id",
          filter_id: "filter_id",
          stage: Stage.YES,
        };
        app.profile.stage.set(data)
          .then((response: any) => {
            const responseWithoutBody = {
              url: response.url,
              options: {
                headers: response.options.headers,
                method: response.options.method
              }
            };
            expect(responseWithoutBody).toMatchSnapshot();
          });
      });

      test("It should call the patch stage endpoint with the the profile reference and filter reference", () => {
        const data: StagePatch = {
          source_id: "source_id",
          profile_id: "profile_id",
          filter_id: "filter_id",
          stage: Stage.YES,
        };
        app.profile.stage.set(data)
          .then((response: any) => {
            const responseWithoutBody = {
              url: response.url,
              options: {
                headers: response.options.headers,
                method: response.options.method
              }
            };
            expect(responseWithoutBody).toMatchSnapshot();
          });
      });

      test("It should call the patch rating endpoint with the the profile id and filter id", () => {
        const data: RatingPatch = {
          source_id: "source_id",
          profile_id: "profile_id",
          filter_id: "filter_id",
          rating: 2
        };
        app.profile.rating.set(data)
          .then((response: any) => {
            const responseWithoutBody = {
              url: response.url,
              options: {
                headers: response.options.headers,
                method: response.options.method
              }
            };
            expect(responseWithoutBody).toMatchSnapshot();
          });
      });

      test("It should call the patch rating endpoint with the the profile id and filter reference", () => {
        const data: RatingPatch = {
          source_id: "source_id",
          profile_id: "profile_id",
          filter_id: "filter_id",
          rating: 2
        };
        app.profile.rating.set(data)
          .then((response: any) => {
            const responseWithoutBody = {
              url: response.url,
              options: {
                headers: response.options.headers,
                method: response.options.method
              }
            };
            expect(responseWithoutBody).toMatchSnapshot();
          });
      });

      test("It should call the patch rating endpoint with the the profile reference and filter id", () => {
        const data: RatingPatch = {
          source_id: "source_id",
          profile_id: "profile_id",
          filter_id: "filter_id",
          rating: 2
        };
        app.profile.rating.set(data)
          .then((response: any) => {
            const responseWithoutBody = {
              url: response.url,
              options: {
                headers: response.options.headers,
                method: response.options.method
              }
            };
            expect(responseWithoutBody).toMatchSnapshot();
          });
      });

      test("It should call the patch rating endpoint with the the profile reference and filter reference", () => {
        const data: RatingPatch = {
          source_id: "source_id",
          profile_id: "profile_id",
          filter_id: "filter_id",
          rating: 2
        };
        app.profile.rating.set(data)
          .then((response: any) => {
            const responseWithoutBody = {
              url: response.url,
              options: {
                headers: response.options.headers,
                method: response.options.method
              }
            };
            expect(responseWithoutBody).toMatchSnapshot();
          });
      });
    });
});