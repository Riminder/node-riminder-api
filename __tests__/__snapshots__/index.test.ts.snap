// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Other tests "Riminder" object relative tests it should throw an error when no key is given 1`] = `"No API Key was supplied for Riminder SDK"`;

exports[`Other tests http module relative tests It should throw an error if the error code is not 200 or 201 1`] = `[Error: An error occured]`;

exports[`Wrapper test Filter endpoints It should call the get filters endpoint 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/filters",
    "path": "/sf/public/api/v1.0/filters",
    "pathname": "/sf/public/api/v1.0/filters",
    "port": null,
    "protocol": "https:",
    "query": null,
    "search": null,
    "slashes": true,
  },
}
`;

exports[`Wrapper test Filter endpoints It sould call the get filter endpoint with the filter id 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/filter?filter_id=filter_id",
    "path": "/sf/public/api/v1.0/filter?filter_id=filter_id",
    "pathname": "/sf/public/api/v1.0/filter",
    "port": null,
    "protocol": "https:",
    "query": "filter_id=filter_id",
    "search": "?filter_id=filter_id",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Filter endpoints It sould call the get filter endpoint with the filter reference 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/filter?filter_reference=filter_reference",
    "path": "/sf/public/api/v1.0/filter?filter_reference=filter_reference",
    "pathname": "/sf/public/api/v1.0/filter",
    "port": null,
    "protocol": "https:",
    "query": "filter_reference=filter_reference",
    "search": "?filter_reference=filter_reference",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile document endpoint with the profile id 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile/documents?source_id=source_id&profile_id=profile_id",
    "path": "/sf/public/api/v1.0/profile/documents?source_id=source_id&profile_id=profile_id",
    "pathname": "/sf/public/api/v1.0/profile/documents",
    "port": null,
    "protocol": "https:",
    "query": "source_id=source_id&profile_id=profile_id",
    "search": "?source_id=source_id&profile_id=profile_id",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile document endpoint with the profile reference 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile/documents?source_id=source_id&profile_reference=profile_reference",
    "path": "/sf/public/api/v1.0/profile/documents?source_id=source_id&profile_reference=profile_reference",
    "pathname": "/sf/public/api/v1.0/profile/documents",
    "port": null,
    "protocol": "https:",
    "query": "source_id=source_id&profile_reference=profile_reference",
    "search": "?source_id=source_id&profile_reference=profile_reference",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile endpoint with the profile id 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile?source_id=source_id&profile_id=profile_id",
    "path": "/sf/public/api/v1.0/profile?source_id=source_id&profile_id=profile_id",
    "pathname": "/sf/public/api/v1.0/profile",
    "port": null,
    "protocol": "https:",
    "query": "source_id=source_id&profile_id=profile_id",
    "search": "?source_id=source_id&profile_id=profile_id",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile endpoint with the profile reference 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile?source_id=source_id&profile_reference=profile_reference",
    "path": "/sf/public/api/v1.0/profile?source_id=source_id&profile_reference=profile_reference",
    "pathname": "/sf/public/api/v1.0/profile",
    "port": null,
    "protocol": "https:",
    "query": "source_id=source_id&profile_reference=profile_reference",
    "search": "?source_id=source_id&profile_reference=profile_reference",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile parsing endpoint with the profile id 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile/parsing?source_id=source_id&profile_id=profile_id",
    "path": "/sf/public/api/v1.0/profile/parsing?source_id=source_id&profile_id=profile_id",
    "pathname": "/sf/public/api/v1.0/profile/parsing",
    "port": null,
    "protocol": "https:",
    "query": "source_id=source_id&profile_id=profile_id",
    "search": "?source_id=source_id&profile_id=profile_id",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile parsing endpoint with the profile reference 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile/parsing?source_id=source_id&profile_reference=profile_reference",
    "path": "/sf/public/api/v1.0/profile/parsing?source_id=source_id&profile_reference=profile_reference",
    "pathname": "/sf/public/api/v1.0/profile/parsing",
    "port": null,
    "protocol": "https:",
    "query": "source_id=source_id&profile_reference=profile_reference",
    "search": "?source_id=source_id&profile_reference=profile_reference",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profiles endpoint 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profiles?source_ids=[source1,source2]&date_start=1234&date_end=1456&page=1&seniority=ALL&filter_id=filter_id&filter_reference=filter_reference&stage=YES&rating=1&limit=30&sort_by=DESC&order_by=RECEPTION",
    "path": "/sf/public/api/v1.0/profiles?source_ids=[source1,source2]&date_start=1234&date_end=1456&page=1&seniority=ALL&filter_id=filter_id&filter_reference=filter_reference&stage=YES&rating=1&limit=30&sort_by=DESC&order_by=RECEPTION",
    "pathname": "/sf/public/api/v1.0/profiles",
    "port": null,
    "protocol": "https:",
    "query": "source_ids=[source1,source2]&date_start=1234&date_end=1456&page=1&seniority=ALL&filter_id=filter_id&filter_reference=filter_reference&stage=YES&rating=1&limit=30&sort_by=DESC&order_by=RECEPTION",
    "search": "?source_ids=[source1,source2]&date_start=1234&date_end=1456&page=1&seniority=ALL&filter_id=filter_id&filter_reference=filter_reference&stage=YES&rating=1&limit=30&sort_by=DESC&order_by=RECEPTION",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the patch rating endpoint with the the profile id and filter id 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-Key": "api_key",
    },
    "method": "PATCH",
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile/rating",
    "path": "/sf/public/api/v1.0/profile/rating",
    "pathname": "/sf/public/api/v1.0/profile/rating",
    "port": null,
    "protocol": "https:",
    "query": null,
    "search": null,
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the patch rating endpoint with the the profile id and filter reference 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-Key": "api_key",
    },
    "method": "PATCH",
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile/rating",
    "path": "/sf/public/api/v1.0/profile/rating",
    "pathname": "/sf/public/api/v1.0/profile/rating",
    "port": null,
    "protocol": "https:",
    "query": null,
    "search": null,
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the patch rating endpoint with the the profile reference and filter id 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-Key": "api_key",
    },
    "method": "PATCH",
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile/rating",
    "path": "/sf/public/api/v1.0/profile/rating",
    "pathname": "/sf/public/api/v1.0/profile/rating",
    "port": null,
    "protocol": "https:",
    "query": null,
    "search": null,
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the patch rating endpoint with the the profile reference and filter reference 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-Key": "api_key",
    },
    "method": "PATCH",
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile/rating",
    "path": "/sf/public/api/v1.0/profile/rating",
    "pathname": "/sf/public/api/v1.0/profile/rating",
    "port": null,
    "protocol": "https:",
    "query": null,
    "search": null,
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the patch stage endpoint with the the profile id and filter id 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-Key": "api_key",
    },
    "method": "PATCH",
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile/stage",
    "path": "/sf/public/api/v1.0/profile/stage",
    "pathname": "/sf/public/api/v1.0/profile/stage",
    "port": null,
    "protocol": "https:",
    "query": null,
    "search": null,
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the patch stage endpoint with the the profile id and filter reference 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-Key": "api_key",
    },
    "method": "PATCH",
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile/stage",
    "path": "/sf/public/api/v1.0/profile/stage",
    "pathname": "/sf/public/api/v1.0/profile/stage",
    "port": null,
    "protocol": "https:",
    "query": null,
    "search": null,
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the patch stage endpoint with the the profile reference and filter id 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-Key": "api_key",
    },
    "method": "PATCH",
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile/stage",
    "path": "/sf/public/api/v1.0/profile/stage",
    "pathname": "/sf/public/api/v1.0/profile/stage",
    "port": null,
    "protocol": "https:",
    "query": null,
    "search": null,
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the patch stage endpoint with the the profile reference and filter reference 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-Key": "api_key",
    },
    "method": "PATCH",
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile/stage",
    "path": "/sf/public/api/v1.0/profile/stage",
    "pathname": "/sf/public/api/v1.0/profile/stage",
    "port": null,
    "protocol": "https:",
    "query": null,
    "search": null,
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the post resume endpoint 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-Key": "api_key",
    },
    "method": "POST",
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/profile",
    "path": "/sf/public/api/v1.0/profile",
    "pathname": "/sf/public/api/v1.0/profile",
    "port": null,
    "protocol": "https:",
    "query": null,
    "search": null,
    "slashes": true,
  },
}
`;

exports[`Wrapper test Source endpoints It should call the get source endpoint 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/source?source_id=id",
    "path": "/sf/public/api/v1.0/source?source_id=id",
    "pathname": "/sf/public/api/v1.0/source",
    "port": null,
    "protocol": "https:",
    "query": "source_id=id",
    "search": "?source_id=id",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Source endpoints It should call the get sources endpoint 1`] = `
Object {
  "options": Object {
    "credentials": "include",
    "headers": Object {
      "X-API-Key": "api_key",
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "www.riminder.net",
    "hostname": "www.riminder.net",
    "href": "https://www.riminder.net/sf/public/api/v1.0/sources",
    "path": "/sf/public/api/v1.0/sources",
    "pathname": "/sf/public/api/v1.0/sources",
    "port": null,
    "protocol": "https:",
    "query": null,
    "search": null,
    "slashes": true,
  },
}
`;
